### PR TITLE
Added odo watch test with debug flag

### PR DESCRIPTION
**Pull request by prietyc123**
**What type of PR is this?**

/kind cleanup

**What does does this PR do / why we need it**:

Adds debug flag tests with odo watch which got removed in https://github.com/openshift/odo/pull/4047 by mistake.

**Which issue(s) this PR fixes**:

Fixes NA

**PR acceptance criteria**:

- [ ] Unit test 

- [x] Integration test 

- [ ] Documentation 

- [x] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:
`make test-cmd-devfile-watch` should pass.

### CODE REVIEW

File: tests/helper/helper_filesystem.go [hunk 1]
```diff
@@ -296,3 +296,15 @@ func ReadFile(filePath string) (string, error) {
...
 	return string(data), nil
 }
+
+// CreateSimpleFile creates a simple file
+// return the file path with random string
+func CreateSimpleFile(context, filePrefix, fileExtension string) (string, string) {
+
+	FilePath := filepath.Join(context, filePrefix+RandString(10)+fileExtension)
+	content := []byte(RandString(10))
+	err := ioutil.WriteFile(FilePath, content, 0644)
```
Comment (mik-dass) [inline tests/helper/helper_filesystem.go:306]: `0644` is throwing security issues https://travis-ci.com/github/openshift/odo/jobs/393570498#L1197

Comment (prietyc123) [reply tests/helper/helper_filesystem.go]: Fixed in https://github.com/openshift/odo/pull/4067/commits/99647241a752fac1029150e953299e6c55d9d9a3

File: tests/integration/devfile/cmd_devfile_watch_test.go [hunk 2]
```diff
@@ -165,6 +165,42 @@ var _ = Describe("odo devfile watch command tests", func() {
 
 	})
 
+	Context("when executing odo watch after odo push with debug flag", func() {
+		It("should be able to start a debug session after push with debug flag using odo watch and revert back after normal push", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", commonVar.Project, cmpName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-debugrun.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
```
Comment (kadel) [inline tests/integration/devfile/cmd_devfile_watch_test.go:173]: if you run `odo create` after this you will save request on devfile registry.
`odo create` can work with the devfile if it is already present in the current directory

File: tests/integration/devfile/cmd_devfile_watch_test.go [hunk 3]
```diff
@@ -165,6 +165,42 @@ var _ = Describe("odo devfile watch command tests", func() {
...
+
+			output := helper.CmdShouldPass("odo", "push", "--project", commonVar.Project)
```
Comment (kadel) [inline tests/integration/devfile/cmd_devfile_watch_test.go:175]: can we remove this as we are pushing again with `debug`  flag right in the next step?

Comment (prietyc123) [reply tests/integration/devfile/cmd_devfile_watch_test.go]: done

File: tests/integration/devfile/cmd_devfile_watch_test.go (hunk 2 above)
Comment (prietyc123) [reply tests/integration/devfile/cmd_devfile_watch_test.go:171]: done

**Comment by mik-dass** [comment]
Failure on Travis CI
```
Running odo with args [odo create nodejs --project esgxoypfov otjcfv]
[odo] I1007 08:20:47.246312   28405 util.go:730] HTTPGetRequest: https://raw.githubusercontent.com/openshift/odo/master/build/VERSION
[odo] I1007 08:20:47.246533   28405 util.go:751] Response will be cached in /tmp/odohttpcache for 1h0m0s
[odo] I1007 08:20:47.246979   28405 util.go:764] Cached response used.
[odo] I1007 08:20:47.261330   28405 util.go:425] path .odo/config.yaml doesn't exist, skipping it
[odo] I1007 08:20:47.261388   28405 util.go:425] path .odo/env/env.yaml doesn't exist, skipping it
[odo] I1007 08:20:47.261422   28405 preference.go:209] The path for preference file is /tmp/902984448/preference.yaml
[odo]  ✗  accepts between 0 and 1 arg when using existing devfile, received 2
```

**Comment by prietyc123** [comment]
> Failure on Travis CI
> 
> ```
> Running odo with args [odo create nodejs --project esgxoypfov otjcfv]
> [odo] I1007 08:20:47.246312   28405 util.go:730] HTTPGetRequest: https://raw.githubusercontent.com/openshift/odo/master/build/VERSION
> [odo] I1007 08:20:47.246533   28405 util.go:751] Response will be cached in /tmp/odohttpcache for 1h0m0s
> [odo] I1007 08:20:47.246979   28405 util.go:764] Cached response used.
> [odo] I1007 08:20:47.261330   28405 util.go:425] path .odo/config.yaml doesn't exist, skipping it
> [odo] I1007 08:20:47.261388   28405 util.go:425] path .odo/env/env.yaml doesn't exist, skipping it
> [odo] I1007 08:20:47.261422   28405 preference.go:209] The path for preference file is /tmp/902984448/preference.yaml
> [odo]  ✗  accepts between 0 and 1 arg when using existing devfile, received 2
> ```

Yes it is happening because of the change request https://github.com/openshift/odo/pull/4067#discussion_r498719438 . AFAIR earlier also I tried creating component after `CopyExample` and it throws arg error. So not sure why is this weird behavior.

**Comment by mik-dass** [comment]
> AFAIR earlier also I tried creating component after CopyExample and it throws arg error. So not sure why is this weird behavior

We don't need to mention the component type in the command

```
[mrinaldas@localhost project]$ odo create some-comp-name
Validation
 ✓  Creating a devfile component from devfile path: /home/mrinaldas/project/devfile.yaml [53396ns]
 ✓  Validating devfile component [104700ns]

Please use `odo push` command to create the component with source deployed
```

**Comment by prietyc123** [comment]
> > AFAIR earlier also I tried creating component after CopyExample and it throws arg error. So not sure why is this weird behavior
> 
> We don't need to mention the component type in the command
> 
> ```
> [mrinaldas@localhost project]$ odo create some-comp-name
> Validation
>  ✓  Creating a devfile component from devfile path: /home/mrinaldas/project/devfile.yaml [53396ns]
>  ✓  Validating devfile component [104700ns]
> 
> Please use `odo push` command to create the component with source deployed
> ```

Then my question is how does `odo create nodejs --project esgxoypfov otjcfv`  works when we create the component before copying devfile from the examples?

**Comment by mik-dass** [comment]
> Then my question is how does odo create nodejs --project esgxoypfov otjcfv works when we create the component before copying devfile from the examples?

It works because there is no devfile.yaml and thus we need to provide the component type to download the appropriate devfile from the registry.

**Comment by prietyc123** [comment]
platform issue.

/retest

**Review by mik-dass — APPROVED** [review]
/lgtm

**Pull request merged**